<!DOCTYPE html>


<html>

<head>
<title>iSwathX</title>

<h1> iSwathX Manual </h1>
</head>

<body>

<p style="font-size:160%;">This manual provides details of functions performed by iSwathX web application (<a href="https://doi.org/10.1093/bioinformatics/bty660" target="_blank"> Noor et al., 2018</a>). It also provides links to the sample files to run. </p>

<li style="font-size:160%;"><a href="#availability">Availability</a></li>
<li style="font-size:160%;"><a href="#libraryfiles">Library Files</a></li>
<li style="font-size:160%;"><a href="#iswathxwizard">iSwathX Wizard</a></li>
<li style="font-size:160%;"><a href="#manualwizard">Manual Wizard</a></li>
<li style="font-size:160%;"><a href="#multilibwizard">MultiLibraries Wizard</a></li>
<li style="font-size:160%;"><a href="#diareportwizard">DIA Report Analysis and Visualization</a></li>
<li style="font-size:160%;"><a href="#references">References</a></li>

<h2><div id="availability"><span style="text-decoration: underline;"><b>1. Availability</b></span></div></h2>
<p style="font-size:160%;">The iSwathX application can be accessed directly from shiny server <a href="https://biolinfo.shinyapps.io/iSwathX/" target="_blank"> iSwathX </a>. For 
more robustness and efficient analysis of large library files, the application can be launched locally through GitHub <a href="https://github.com/znoor/iSwathX" target="_blank">iSwathX</a> 
using following commands in R: 
   <li style="font-size:160%;"> > install.packages("shiny")</li>
   <li style="font-size:160%;"> > library(shiny)</li>
   <li style="font-size:160%;"> > shiny::runGitHub("iSwathX", "znoor")</li>
</p>

<br></br>

<p style="font-size:160%;"> All the functions performed by the application are also available in the form of an R package. The package can be downloaded and installed from GitHub using following commands in R:

   <li style="font-size:160%;"> > install.packages("devtools")</li>
   <li style="font-size:160%;"> > library(devtools)</li>
   <li style="font-size:160%;"> > install_github("znoor/iSwathX", subdir = "package")</li>
   
   <li style="font-size:160%;"> > rm(list = ls())</li>
   <li style="font-size:160%;"> > library(iSwathX)</li>

</p>

<h3><i>1.1. Home Page</i></h3>
<p style="font-size:160%;">Home page of iSwathX application presents the work flow for the generation and integration of libraries. For each step of library generation, 
external web links for tools/software have been given. It provides an overview and a starting point for library analysis from raw mass spectrometry files.
Further, library integration work flow shows the two modules of the iSwathX application.</p>

<hr></hr>

<h2><div id="libraryfiles"><span style="text-decoration: underline;"><b>2. Library Files</b></span></div></h2>
<p style="font-size:160%;">The iSwathX application can read and analyze assay library files in four different formats.</p>

<li style="font-size:160%;">PeakView</li>
<li style="font-size:160%;">OpenSWATH</li>
<li style="font-size:160%;">Skyline</li>
<li style="font-size:160%;">Spectronaut</li>


<p style="font-size:160%;">These assay libraries have different column names to make them compatible with the corresponding data analysis software. Details of 
these column names are provided in <a href="https://www.nature.com/articles/nprot.2015.015" target="_blank"> Schubert et al. </a> <a href="https://media.nature.com/original/nature-assets/nprot/journal/v10/n3/extref/nprot.2015.015-S1.pdf" target="_blank"> 
(Supplementary file)</a> . However, to perform the analysis, all libraries should have following information: 
<li style="font-size:160%;">Precursor m/z</li>
<li style="font-size:160%;">Fragment m/z</li>
<li style="font-size:160%;">Retention times</li>
<li style="font-size:160%;">Relative intensity</li>
<li style="font-size:160%;">Peptide sequences without modifications</li>
<li style="font-size:160%;">Peptide sequences with modifications</li>
<li style="font-size:160%;">Precursor charge</li>
<li style="font-size:160%;">Protein name</li>
<li style="font-size:160%;">Fragment type</li>
<li style="font-size:160%;">Fragment charge</li>
<li style="font-size:160%;">Fragment ion number</li>
<li style="font-size:160%;">Uniprot ID</li>
<li style="font-size:160%;">Isotype type (optional)</li>


</p>
<p style="font-size:160%;">
Example library files are provided here in different formats (smaller versions), which can be downloaded and tested. Details of each of the library files can be acquired from 
<a href="http://www.mcponline.org/content/15/7/2501.full" target="_blank"> Wu et al., 2016</a>.

<li style="font-size:160%;"> <a href="libraries/Lib1_peakview.csv" download target="_blank">Lib1_peakview.csv</a></li>
<li style="font-size:160%;"> <a href="libraries/Lib2_openswath.csv" download target="_blank">Lib2_openswath.csv</a></li>
<li style="font-size:160%;"> <a href="libraries/Lib3_skyline.csv" download target="_blank">Lib3_skyline.csv</a></li>
<li style="font-size:160%;"> <a href="libraries/Lib6_spectronaut.csv" download target="_blank">Lib6_spectronaut.csv</a></li>
<li style="font-size:160%;"> <a href="libraries/Lib2_3_peakview.csv" download target="_blank">Lib2_3_peakview.csv</a></li>

</p>

<hr></hr>
<h2><div id="iswathxwizard"><span style="text-decoration: underline;"><b>3. iSwathX Wizard</b></span></div></h2>

<p style="font-size:160%;">This is a fully automatic wizard for performing the library conversion and integration tasks. It takes input files and performs the analysis and integration using 
default parameters.</p>

<h3><i>3.1. Input Libraries</i></h3>
<p style="font-size:160%;">This section takes input files in desired format using interactive upload facility. 
</p>

<h3><i>3.2. Parameter Used</i></h3>
<p style="font-size:160%;">This section shows all the default parameters for quality checks and library combinations. 
</p>

<h3><i>3.3. Log</i></h3>
<p style="font-size:160%;">This section shows the brief summary of input and output library files. 
</p>

<h3><i>3.4. Library Table</i></h3>
<p style="font-size:160%;">Combined library is displayed in the form of a table. 
</p>

<h3><i>3.5. Download Library</i></h3>
<p style="font-size:160%;">Combined library can be downloaded in desired format using interactive download facility. 
</p>

<h3><i>3.6. Video Tutorial</i></h3>
<p style="font-size:160%;">A video tutorial for the working of this module is provided here. 
<video width = "80%" height = "80%" controls>
<source src = "video-tutorials/iSwathWizard.mp4" type = "video/mp4">
</video>
</p>

<hr></hr>

<h2><div id="manualwizard"><span style="text-decoration: underline;"><b>4. Manual Wizard</b></span></div></h2>

<h3><b>4.1. Libraries Input and Format Conversion</b></h3>

<h3><i>4.1.1. Input Libraries</i></h3>
<p style="font-size:160%;">This section takes input files in desired format using interactive upload facility. 
</p>

<h3><i>4.1.2 Cleaning Libraries</i></h3>
<p style="font-size:160%;">This section filters the library file by:
<li style="font-size:160%;"> Removing low confident peptides</li>
<li style="font-size:160%;"> Removing low intensity ions</li>
<li style="font-size:160%;"> Removing peptides with modifications for missed cleavages.</li> 
</p>

<h3><i>4.1.3. Download Library</i></h3>
<p style="font-size:160%;">Seed library and addon library can be downloaded in desired format using interactive download facility. 
</p>

<h3><i>4.1.4. Video Tutorial</i></h3>
<p style="font-size:160%;">A video tutorial for the working of this module is provided here. 
<video width = "80%" height = "80%" controls>
<source src = "video-tutorials/LibraryInputOutput.mp4" type = "video/mp4">
</video>
</p>

<h3><b>4.2. Libraries Comparison</b></h3>

<h3><i>4.2.1. Retention Time and Relative Intensity Correlation</i></h3>
<p style="font-size:160%;">This section uses the input seed and addon library (uploaded previously) and performs retention time correlation using either the retention times of both 
libraries or retention times and hydrophobicity index of libraries, and relative intensity correlation using intensity values of both libraries.

As a result, it displays:
<li style="font-size:160%;"> The data table of training set (common peptides of both libraries), their retention times, hydrophobicity indexes and relative intensity values</li>
<li style="font-size:160%;"> Correlation graphs</li>
<li style="font-size:160%;"> Residual plot for the selected regression model</li>
</p>

<h3><i>4.2.2. Video Tutorial</i></h3>
<p style="font-size:160%;">A video tutorial for the working of this module is provided here. 
<video width = "80%" height = "80%" controls>
<source src = "video-tutorials/LibraryComparison.mp4" type = "video/mp4">
</video>
</p>

<h3><b>4.3. Building Combined Libraries</b></h3>

<h3><i>4.3.1. Parameters</i></h3>
<p style="font-size:160%;">This section uses the input seed and add-on library (uploaded previously), provides different parameters to be applied while combining the seed and addon libraries 
and performs integration.
<li style="font-size:160%;"> Method : method for aligning and predicting the retention times using either 'time' or 'hydrophobicity' methods.</li>
<li style="font-size:160%;"> Consolidate Accessions : to consolidate the protein accessions in base and integrated library.</li>
<li style="font-size:160%;"> Generate Plots : to plot the libraries statistics and analysis.</li>
<li style="font-size:160%;"> Merge Libraries : to merge the base library to the new spectra of addon library.</li>
<li style="font-size:160%;"> Re-calibrate Retention Times : To adjust the retention times of addon library in case of low retention times correlation.</li>
<li style="font-size:160%;"> Retention Time Correlation Cutoff : Minimum correlation coefficient.</li>
<li style="font-size:160%;"> Training size Cutoff :  Minimum number of common peptides in training set.</li>
</p>

<h3><i>4.3.2. Library Table</i></h3>
<p style="font-size:160%;">Combined library is displayed in the form of a table. 
</p>

<h3><i>4.3.3. Download Library</i></h3>
<p style="font-size:160%;">Combined library can be downloaded in desired format using interactive download facility. 
</p>

<h3><i>4.3.4. Video Tutorial</i></h3>
<p style="font-size:160%;">A video tutorial for the working of this module is provided here. 
<video width = "80%" height = "80%" controls>
<source src = "video-tutorials/LibraryIntegration.mp4" type = "video/mp4">
</video>
</p>

<h3><b>4.4. Data Visualization</b></h3>

<h3><i>4.4.1. Correlation and Library Stats Plots</i></h3>
<p style="font-size:160%;">In this module, all the graphs generated during processing can be visualized altogether and can be saved. These graphs include: 
<li style="font-size:160%;"> Retention time correlation</li>
<li style="font-size:160%;"> Correlation residuals</li>
<li style="font-size:160%;"> Intensity correlation</li>
<li style="font-size:160%;"> Histograms of peptides and ions in base and addon libraries</li>
<li style="font-size:160%;"> Density plots of base and addon libraries</li>
<li style="font-size:160%;"> Peptides and protein numbers in base and addon libraries</li>
<li style="font-size:160%;"> Peptides and protein Venn diagrams for base and addon libraries</li>
<li style="font-size:160%;"> Library coverage plot in base and combined libraries</li>
</p>

<h3><i>4.4.2. Video Tutorial</i></h3>
<p style="font-size:160%;">A video tutorial for the working of this module is provided here. 
<video width = "80%" height = "80%" controls>
<source src = "video-tutorials/Visualization.mp4" type = "video/mp4">
</video>
</p>

<hr></hr>

<!--Adding for new module-->

<h2><div id="multilibwizard"><span style="text-decoration: underline;"><b>5. MultiLibraries Wizard</b></span></div></h2>

<h3><b>5.1. Reading and Comparing Multiple Libraries</b></h3>

<h3><i>5.1.1. Reading Multiple Libraries</i></h3>
<p style="font-size:160%;">This section takes input files in desired format using interactive upload facility. User can select one of the two given options, either 'Reading Three Libraries' or 'Reading Four Libraries'. 
</p>

<h3><i>5.1.2 Libraries Summary</i></h3>
<p style="font-size:160%;">This section shows the brief summary of input library files:
<li style="font-size:160%;"> Number of proteins</li>
<li style="font-size:160%;"> Number of peptides</li>
<li style="font-size:160%;"> Number of transition ions</li> 
</p>

<h3><i>5.1.3. Libraries Comparison</i></h3>
<p style="font-size:160%;">This section uses the input seed and add-on libraries (uploaded previously) and performs retention time correlation and relative intensity correlation of seed lib with each of the add-on library, and rank them accordingly.
As a result, it displays:
<li style="font-size:160%;"> The data table of common peptides, retention time correlation, intensity correlation and rank of each library on the basis of correlation values. Add-on libraries are merged to the base library in a sequential manner according to their rank.</li>
<li style="font-size:160%;"> Correlation graphs</li>
</p>

<h3><i>5.1.4. Video Tutorial</i></h3>
<p style="font-size:160%;">A video tutorial for the working of this module is provided here. 
<video width = "80%" height = "80%" controls>
<source src = "video-tutorials/MultiLibraryRead_Comparison.mp4" type = "video/mp4">
</video>
</p>

<h3><b>5.2. Building Combined Libraries</b></h3>

<h3><i>5.2.1. Parameters</i></h3>
<p style="font-size:160%;">This section uses the input seed and add-on libraries (uploaded previously), provides different parameters to be applied while combining the seed and add-on libraries 
and performs integration.
<li style="font-size:160%;"> Method : method for aligning and predicting the retention times using either 'time' or 'hydrophobicity' methods.</li>
<li style="font-size:160%;"> Consolidate Accessions : to consolidate the protein accessions in base and integrated library.</li>
<li style="font-size:160%;"> Generate Plots : to plot the libraries statistics and analysis.</li>
<li style="font-size:160%;"> Merge Libraries : to merge the base library to the new spectra of addon library.</li>
<li style="font-size:160%;"> Re-calibrate Retention Times : To adjust the retention times of addon library in case of low retention times correlation.</li>
<li style="font-size:160%;"> Retention Time Correlation Cutoff : Minimum correlation coefficient.</li>
<li style="font-size:160%;"> Training size Cutoff :  Minimum number of common peptides in training set.</li>
</p>

<h3><i>5.2.2. Library Table</i></h3>
<p style="font-size:160%;">Combined library is displayed in the form of a table. 
</p>

<h3><i>5.2.3. Download Library</i></h3>
<p style="font-size:160%;">Combined library can be downloaded in desired format using interactive download facility. 
</p>

<h3><i>5.2.4. Video Tutorial</i></h3>
<p style="font-size:160%;">A video tutorial for the working of this module is provided here. 
<video width = "80%" height = "80%" controls>
<source src = "video-tutorials/MultiLibraryIntegration.mp4" type = "video/mp4">
</video>
</p>

<h3><b>5.3. Data Visualization</b></h3>

<h3><i>5.3.1. Venn Diagrams</i></h3>
<p style="font-size:160%;">In this module, Venn diagrams of proteins and peptides are shown for each iteration (merging of each add-on library). 
</p>

<h3><i>5.3.2. Video Tutorial</i></h3>
<p style="font-size:160%;">A video tutorial for the working of this module is provided here. 
<video width = "80%" height = "80%" controls>
<source src = "video-tutorials/MultiVisualization.mp4" type = "video/mp4">
</video>
</p>

<hr></hr>

<!--Adding for new module-->

<h2><div id="diareportwizard"><span style="text-decoration: underline;"><b>6. DIA Report Analysis and Visualization</b></span></div></h2>

<h3><b>6.1. DIA Report File Reading and Processing</b></h3>

<h3><i>6.1.1. Reading DIA Report File</i></h3>
<p style="font-size:160%;">This section reads the DIA report file (currently from the Skyline Software). 
</p>

<h3><i>6.1.2 Process Report File</i></h3>
<p style="font-size:160%;">This section process the report file by:
<li style="font-size:160%;"> Removing standard peptides</li>
<li style="font-size:160%;"> Calculating average dotp values</li>
<li style="font-size:160%;"> Calculating average mass error values</li>
<li style="font-size:160%;"> Calculating average retention time values</li>
<li style="font-size:160%;"> Calculating average q-values values</li> 
</p>

<h3><i>6.1.3. Download Report File</i></h3>
<p style="font-size:160%;">Processed report file (with added columns) can be downloaded using interactive download facility. 
</p>

<h3><i>6.1.4. Video Tutorial</i></h3>
<p style="font-size:160%;">A video tutorial for the working of this module is provided here. 
<video width = "80%" height = "80%" controls>
<source src = "video-tutorials/ReadingReportFile.mp4" type = "video/mp4">
</video>
</p>

<h3><b>6.2. Quality Check Plots</b></h3>

<h3><i>6.2.1. Plotting different quality parameters among data from runs</i></h3>
<p style="font-size:160%;">This section uses the input report file (uploaded previously) and plots different parameters to be applied performing DIA data analysis.
<li style="font-size:160%;"> Dot Product : a correlation between observed (library) and measured (DIA) spectra and ‘1’ means a perfect match.</li>
<li style="font-size:160%;"> Intensity Distribution : Area under the curve for peptides.</li>
<li style="font-size:160%;"> Mass Error : to plot the libraries statistics and analysis.</li>
<li style="font-size:160%;"> Coefficient of Variation (CV) (among technical replicates) : the ratio of peptide abundance to the median peptide abundance measured across all samples. </li>
<li style="font-size:160%;"> Q-Values : the minimum false discovery rate (FDR).</li>
<li style="font-size:160%;"> DDA-DIA RT Correlation : Pearson correlation between peptide retention times from DDA and DIA datasets.</li>
</p>

<h3><i>6.2.2. Video Tutorial</i></h3>
<p style="font-size:160%;">A video tutorial for the working of this module is provided here. 
<video width = "80%" height = "80%" controls>
<source src = "video-tutorials/PlottingReportFile.mp4" type = "video/mp4">
</video>
</p>

<h3><b>6.3. Proteins Intensity</b></h3>

<h3><i>6.3.1. Calculating and Plotting Protein Intensity Values </i></h3>
<p style="font-size:160%;">This section uses the input report file (uploaded previously), and plots protein intensities calculated by summing the normalized area values of corresponding peptides.
</p>

<h3><i>6.3.2. Video Tutorial</i></h3>
<p style="font-size:160%;">A video tutorial for the working of this module is provided here. 
<video width = "80%" height = "80%" controls>
<source src = "video-tutorials/PlottingProtIntensities.mp4" type = "video/mp4">
</video>
</p>

<h3><b>6.4. Multiple Reports Comparison</b></h3>

<h3><i>6.4.1. Reading Report Files</i></h3>
<p style="font-size:160%;">This section reads report files (currently in Skyline format) using interactive upload facility. User is asked to set the names for each of the datasets. 
</p>

<h3><i>6.4.2. Comparison Plots</i></h3>
<p style="font-size:160%;">This section performs and plots the comparison between multiple report files. 

It displays:
<li style="font-size:160%;"> Report Statistics: Number of proteins and peptides in each dataset</li>
<li style="font-size:160%;"> Venn Diagrams: Common and unique proteins and peptides quantified in each dataset</li>
<li style="font-size:160%;"> Retention Time Correlation: DIA Retention time correlation between common peptides quantified in each dataset</li>
<li style="font-size:160%;"> Coefficient of Variation (CV): CV (the ratio of peptide abundance to the median peptide abundance measured across all samples) comparison</li>
</p>

<h3><i>6.4.3. Video Tutorial</i></h3>
<p style="font-size:160%;">A video tutorial for the working of this module is provided here. 
<video width = "80%" height = "80%" controls>
<source src = "video-tutorials/MultiReportVisualization.mp4" type = "video/mp4">
</video>
</p>

<hr></hr>

<h2><div id="references"><span style="text-decoration: underline;">References</span></div></h2>

<li> Noor, Zainab., et al. "iSwathX: an interactive web-based application for extension of DIA peptide reference libraries." Bioinformatics (2018).
<li> Schubert, Olga T., et al. "Building high-quality assay libraries for targeted analysis of SWATH MS data." Nature protocols 10.3 (2015): 426.
<li> Wu, Jemma X., et al. "SWATH mass spectrometry performance using extended peptide MS/MS assay libraries." Molecular & Cellular Proteomics 15.7 (2016): 2501-2514.

</body>
</html>